python/cuspatial/cuspatial/core/binops/binops.py

```diff
@@ -0,0 +1,409 @@
+            self.rhs
+        ):
+            df_result = result.groupby("idx").sum().sort_index() > 0
+        else:
+            df_result = result.groupby("idx").sum().sort_index() > 1
+        point_result = cudf.Series(
+            df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
+        )
+        point_result.name = None
+        return point_result
+
+
+class IntersectsBinop(ContainsProperlyBinop):
+    def preprocess(self, op, lhs, rhs):
+        if contains_only_polygons(rhs):
+            (lhs, rhs) = (rhs, lhs)
+        return super().preprocess(op, lhs, rhs)
+
+    def postprocess(self, op, point_indices, point_result):
+        """Postprocess the output GeoSeries to ensure that they are of the
+        correct type for the operation."""
+        return super().postprocess(op, point_indices, point_result)
+
+
+class WithinBinop(ContainsProperlyBinop):
+    def preprocess(self, op, lhs, rhs):
+        if contains_only_polygons(rhs):
+            (lhs, rhs) = (rhs, lhs)
+        return super().preprocess(op, lhs, rhs)
+
+    def postprocess(self, op, point_indices, point_result):
+        """Postprocess the output GeoSeries to ensure that they are of the
+        correct type for the operation."""
+        result = cudf.DataFrame({"idx": point_indices, "pip": point_result})
+        df_result = result
+        # Discrete math recombination
+        if (
+            contains_only_linestrings(self.rhs)
+            or contains_only_polygons(self.rhs)
+            or contains_only_multipoints(self.rhs)
+        ):
+            # process for completed linestrings, polygons, and multipoints.
+            # Not necessary for points.
+            df_result = (
+                result.groupby("idx").sum().sort_index()
+                == result.groupby("idx").count().sort_index()
+            )
+        point_result = cudf.Series(
+            df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
+        )
+        point_result.name = None
+        if op == "equals":
+            if len(point_result) == 1:
+                return point_result[0]
+        return point_result
+
+
+class CrossesBinop(_binop):
+    def preprocess(self, op, lhs, rhs):
+        return super().preprocess(op, lhs, rhs)
+
+    def postprocess(self, op, point_indices, point_result):
+        result = cudf.DataFrame({"idx": point_indices, "pip": point_result})
+        df_result = result
+        # Discrete math recombination
+        if (
+            contains_only_linestrings(self.rhs)
+            or contains_only_polygons(self.rhs)
+            or contains_only_multipoints(self.rhs)
+        ):
+            df_result = ~result
+        point_result = cudf.Series(
+            df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
+        )
+        point_result.name = None
+        return point_result
+
+
+class EqualsBinop(_binop):
+    def preprocess(self, op, lhs, rhs):
+        return super().preprocess(op, lhs, rhs)
```

Comment:
```suggestion
```
If it does nothing, you don't need to write it :)

Reply:
In that case it had to be written because it is an `@abstractmethod` and has to have one in order to inherit from `_binop(ABC)`. For now I've refactored it to inherit from `ContainsProperlyBinpred` instead, but I will iterate it again, especially in the next PRs as it gets implemented more thoroughly.